lib/engineCustomConditions.js

```diff
@@ -0,0 +1,262 @@
+/* global -Promise */
+'use strict';
+
+var utils = require('./utils'),
+    _ = require('underscore'),
+    Promise = require('bluebird');
+
+var EngineCustomConditions = (function() {
+    return function() {
+
+        /* hmda-syntactical */
```

Comment:
Do comments like these actually add anything anymore? When everything was in single file they did help delineate between different areas, but now that you've modularized the code it seems like they should be replaced by jsdoc comments on each function. If so, that might be part of the system documentation task.